src/test/java/org/jabref/logic/importer/ImportFormatReaderIntegrationTest.java

```diff
@@ -55,6 +55,20 @@ public void testImportFormatFromFile() throws Exception {
         assertEquals(count, reader.importFromFile(format, file).getDatabase().getEntries().size());
     }
 
+    @Test
+    public void testImportUnknownFormatFromString() throws Exception {
+        String data = new String(Files.readAllBytes(file), StandardCharsets.UTF_8);
+        try {
+            assertEquals(count, reader.importUnknownFormatFromString(data).parserResult.getDatabase().getEntries().size());
+        } catch (ImportException e) {
+            // msbib test file does not contain an author
+            if ("msbib".equals(format)) {
```

Comment:
Why is this special treatment necessary for the import from a string, but not when the same content is imported from a file?